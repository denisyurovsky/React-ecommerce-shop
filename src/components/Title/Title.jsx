import propTypes from 'prop-types';
import React from 'react';

import styles from './Title.module.scss';

export const Title = ({ children }) => {
  return (
    <h2 data-testid="title" className={styles.header}>
      {children}
    </h2>
  );
};

Title.propTypes = {
  children: propTypes.string,
};
