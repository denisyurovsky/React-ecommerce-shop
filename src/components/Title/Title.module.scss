@import "../../assets/styles/colors.scss";

.header {
  color: $primaryMain;
  font-size: 40px;
  font-weight: 400;
}

@media all and (max-width: 700px) {
  .header {
    font-size: 26px;
  }
}
