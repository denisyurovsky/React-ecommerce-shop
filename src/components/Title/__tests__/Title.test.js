import { render } from '@testing-library/react';
import React from 'react';

import { Title } from '../Title';

describe('Title component', () => {
  describe('snapshots', () => {
    it('renders a valid snapshot', () => {
      const { asFragment } = render(<Title>Title text</Title>);

      expect(asFragment()).toMatchSnapshot();
    });
  });
});
