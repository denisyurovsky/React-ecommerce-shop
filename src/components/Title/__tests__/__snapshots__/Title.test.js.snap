// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Title component snapshots renders a valid snapshot 1`] = `
<DocumentFragment>
  <h2
    class="header"
    data-testid="title"
  >
    Title text
  </h2>
</DocumentFragment>
`;
